aws/resource_aws_dms_endpoint_test.go

```diff
@@ -142,7 +144,8 @@ func TestAccAwsDmsEndpoint_Elasticsearch(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					checkDmsEndpointExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "elasticsearch_settings.#", "1"),
-					resource.TestCheckResourceAttr(resourceName, "elasticsearch_settings.0.endpoint_uri", "search-estest.us-west-2.es.amazonaws.com"),
+					resource.TestMatchResourceAttr(resourceName, "elasticsearch_settings.0.endpoint_uri", regexp.MustCompile(
+						`^search\-estest\.[^\.]+.es.`)),
```

Comment:
We can potentially leverage `testAccMatchResourceAttrRegionalHostname` here or create a `testAccCheckResourceAttrRegionalHostname` equivalent 👍  e.g.

```go
// testAccCheckResourceAttrRegionalHostname ensures the Terraform state exactly matches a formatted DNS hostname with region and partition DNS suffix
func testAccCheckResourceAttrRegionalHostname(resourceName, attributeName, serviceName string, hostnamePrefix string) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		hostname := fmt.Sprintf("%s.%s.%s.%s", hostnamePrefix, serviceName, testAccGetRegion(), testAccGetPartitionDNSSuffix())

		return resource.TestCheckResourceAttr(resourceName, attributeName, hostname)(s)
	}
}
```

Although it also looks like this hostname is currently PREFIX.REGION.SERVICE.DOMAIN -- since its a fake hostname maybe we should make it look like the format required for these functions anyways. This is what we get for the API not validating things upfront and us utilizing that lack of validation to not spin up a real Elasticsearch Domain to use the attribute reference. 😛